### ISSUE TITLE
Manuscript cannot be deleted.

### ISSUE THREAD

**Issue by ghost** [worldanvil/worldanvil-bug-tracker#206]
**World Anvil Username**: KajetanWrites

**Manuscript Title**: Critique

**Feature**: Manuscript

**Describe the Issue**
I have a manuscript called "Critique" which I meant to use to post parts of other manuscripts to get some feedback, but I feel it wasn't such a good idea and want to delete it. When I try to do that everything is fine at first, I type the exact title and press delete, but then I get 404 and when I get back to WA the manuscript is still there.

**Expected behavior**
Manuscript deleted

**Comment by iamromeo** [comment]
I will need more details about the state of the manuscript - as is it cannot be reproduced.

**Comment by KajetanWrites** [comment]
@iamromeo What do you need? Manuscript is published

**Comment by Kordishal** [comment]
![image](https://user-images.githubusercontent.com/14262536/116365612-b599f200-a805-11eb-9737-48a8bcf9cef4.png)

The delete fails due to a form error most likely. Or there is an unexpected structure. I can't really figure out what is going on. The easiest way would probably be if you were to transfer the Manuscript to your account and make a copy of it.

https://www.worldanvil.com/homer/manuscript/delete/12612126-21d2-4c00-89a7-c8481642759a

**Comment by Kordishal** [comment]
If this doesn't work I suggest removing content from the parts and maybe moving them to the trash. Or removing clearing configurations.

**Comment by Kordishal** [comment]
Were you able to delete the manuscript? Or does it still exist?

**Comment by KajetanWrites** [comment]
Sorry, I completely forgot about this issue. I will check in a moment

**Comment by KajetanWrites** [comment]
still not working

**Comment by polymorphic2346** [comment]
> still not working

I had this problem, too, on a draft I created before manuscripts were in open beta. I finally got it to delete by removing everything from the bin folder, deleting the text in each page, then using the "delete manuscript" and getting the 404 until there was only one single page left in the manuscript. (it seemed to delete the sections one by one). Then I created a new page, added some characters with a short keysmash, moved the old one to the bin, and used "delete manuscript." That deleted the final old page that was in the bin. But it was still 404ing. I then moved the new page to the bin. And FINALLY the whole thing was able to be deleted.

TLDR: Remove everything from the bin, mess around with the file structure, maybe put some things back in the bin, it will come back like it's haunted until it's deleted piece by piece.

**Comment by SerenPedac** [comment]
I'm having this issue as well. The solution to go back and keep deleting stuff won't work for me because the manuscript is "deleted" but it is still showing up when I click on "manuscripts".

**Comment by PrincessHoneytea** [comment]
I'm also having this issue, with two "deleted" manuscripts. Both are private, unpublished, attached to a private world. One has about 200 words in it, one has about 30 000 words in it. I can't delete them at all and I can't access them at all. I get 404 pages when I try. Thanks for all that you all do!

**Comment by Amancham** [comment]
Here is what I noticed: 

I have a lifetime Grandmaster Account (in case that makes a difference), username Amancham
Normal non-beta account (no alchemists council for me just yet)
I am working on Firefox Browser, latest release.
Stable, fast internet connection via cable (no Wireless to blame) 

### My study opject:
![1_pre-delete](https://user-images.githubusercontent.com/25498928/130229466-5d4dd149-14d4-44de-a5f3-8bc10aa09b67.png)

* Unpublished manuscript
* set to private (Just freshly created and all left at default values)
* Changed name to Bughunt

### Steps taken and problems encountered:

1) Go to manuscript list 
2) on the Bughunt manuscript hit the delete button 
3) on the next page enter "Bughunt" as the name, hit the button, confirm the popup 

I am forwarded to a 404 error page. I manually changed the url to go back to the list and found the manuscript still there but: 
![2_after-delete](https://user-images.githubusercontent.com/25498928/130229488-b1ada784-35aa-43f1-a018-37f4c8b4f5d3.png)

* Manuscript folder (with the Name Bughunt) was gone 
* Instead I had a bold "Chapter 1" folder and another Chapter one with Scene 1 from the initial set up
* Research folder and contents as well as bin are still there 

So I moved everything into the bin and tried deleting again. Error 404 again.
![3_all-in-bin](https://user-images.githubusercontent.com/25498928/130229520-74d9df19-a758-4370-b22e-6a008ad2ccdb.png)

Again I get a new "Chapter 1" folder in bold above the bin but the exact same bold written folder is gone from the bin only Research Folder, Chapter 1 folder and the files "Interesting Links"  and "Scene 1" remain in the bin. 
![4_after-delete2](https://user-images.githubusercontent.com/25498928/130229538-e900dfa5-7b60-4db1-8b90-8a12021eca04.png)

I have tried several times to delete and at some point it deleted the contents of the bin but not the bin or the bold chapter 1 folder itself. After many, many tries of moving stuff to bin in different ways, keeping bin open or closed and such, the manuscript was deleted. But honestly, I stopped recording the steps after several unsuccessful tries to delete.

I have tried the same with a manuscript that I published, but still kept private, and the same things happened. It only deletes the manuscript folder itself but not the contents of the folder. All of those are set to the toplevel no matter if they are folders or textfiles but since the actual manuscript or main folder is then missing, it's all messed up and funky. 

### Workaround: 

What I have found works best in terms of getting the manuscript GONE is: 
* delete it and get to the 404 error page 
* hit the "one step back" button on the browser to go back to the page where you have to enter the manuscript name and confirm that you really want to delete 
* just hit the button again and confirm in the popup again. (the name should still be in there) 
* Then the manuscript is gone. 

Do NOT open it up in between. I had runs where I tried 10 times to delete, moved things in and out of the bin and it just would not do it. If you hit delete, open the manuscript after the 404 error and then try to delete again, it will just come up with the 404 error every time.

I also had an older Manuscript that failed to delete, the same "go back and delete it twice in a row" trick worked on that as well. Though on another run with a manuscript I marked as "completed" I had to do the go back and do it again trick 3 times to get rid of the manuscript 

Since I have only recently upgraded, I have no old manuscripts to see if the workaround also works on older stuff, unfortunately. 

I don't know if that helps at all, but I figured I write it up since I was struggling with the same 'can't delete it' problem. I don't want to mess around too much, so I quit after a few attempts and bughunt-manuscripts. Sorry for the long message, but I hope this is easier to reproduce since somehwere further up it was not possible to reproduce the error?

**Sidenote:** 

Noticed something else, that I found a little odd. Positition of the research folder changes on newly created manuscripts. Sometimes when I open the manuscript for the first time, the research folder is the top most folder and other times I open a new manuscript for the first time and the research folder is between the manuscript folder and the bin
Not sure if that warrants a whole new bug report, though.

Also noticed the collapse/expand on the bin not working as expected, but might make a seperate bugreport for that one.

**Comment by Kordishal** [comment]
So I figured out at least one reason as to why manuscripts can't be deleted:

When the world they were created in is deleted the manuscript continue to exist, but loses the root element. This means the manuscript can no longer be deleted.

**Comment by iamromeo** [comment]
> So I figured out at least one reason as to why manuscripts can't be deleted:
> 
> When the world they were created in is deleted the manuscript continue to exist, but loses the root element. This means the manuscript can no longer be deleted.

Well that is a very interesting point.

**Comment by KajetanWrites** [comment]
In my case the world the manuscript was attached to wasn't deleted

**Comment by Amancham** [comment]
> In my case the world the manuscript was attached to wasn't deleted

Same here. I did several trial runs and the world was not deleted.

**Comment by JohnHFriez** [comment]
I have the identical issues.  I have not deleted any worlds.  The MS is eternal.

**Comment by audrey-jensen** [comment]
I have this issue with 3 separate manuscripts, and no matter how many times I try to delete them, I get the 404 and the manuscript isn't deleted from the manuscripts list.

**Comment by DerangedDM** [comment]
Issue persists. I have 5 manuscripts I can't kill.

**Comment by Kordishal** [comment]
Closing this issue as it has been reported on the new system. Upvote the issue there for visibility:

https://www.worldanvil.com/udan-tracker/c47d0b49-9cdb-4d6a-884c-5b4fbe9421a9/view
